frontend/public/components/app-contents.tsx

```diff
@@ -80,21 +80,57 @@ const LazyRoute = (props) => (
   />
 );
 
-const getPluginPageRoutes = (activePerspective: string, routePages: RoutePage[]) =>
-  routePages.map((r) => {
-    if (r.properties.perspective && r.properties.perspective !== activePerspective) {
-      return null;
-    }
-    const Component = r.properties.loader ? LazyRoute : Route;
-    return <Component {...r.properties} key={Array.from(r.properties.path).join(',')} />;
-  });
+const getPluginPageRoutes = (
+  activePerspective: string,
+  setActivePerspective: (perspective: string) => void,
+  routePages: RoutePage[],
+) => {
+  const activeRoutes = routePages
+    .filter((r) => !r.properties.perspective || r.properties.perspective === activePerspective)
+    .map((r) => {
+      const Component = r.properties.loader ? LazyRoute : Route;
+      return <Component {...r.properties} key={Array.from(r.properties.path).join(',')} />;
+    });
+
+  const inactiveRoutes = routePages
+    .filter((r) => r.properties.perspective && r.properties.perspective !== activePerspective)
+    .map((r) => {
+      const key = Array.from(r.properties.path).join(',');
```

Comment:
Include the perspective as part of the `key`.

Reply:
Added.